ui/components/component-library/avatar-token/avatar-token.types.ts

```diff
@@ -0,0 +1,55 @@
+import { AvatarBaseSize } from '../avatar-base';
+import { ValidTag } from '../text';
+import type { BoxProps } from '../../ui/box/box.d';
+import {
+  BackgroundColor,
+  BorderColor,
+  TextColor,
+} from 'ui/helpers/constants/design-system';
+
+/**
+ * Props for the AvatarToken component
+ */
+export interface AvatarTokenProps extends BoxProps {
+  /**
+   * The name accepts the string to render the first letter of the AvatarToken. This will be used as the fallback display if no image url is passed to the src
+   */
+  name?: string;
+  /**
+   * The src accepts the string of the image to be rendered
+   */
+  src?: string;
+  /**
+   * The showHalo accepts a boolean prop to render the image with halo effect
+   */
+  showHalo?: boolean;
+  /**
+   * The size of the AvatarToken.
+   * Possible values could be 'AvatarBaseSize.Xs' 16px, 'AvatarBaseSize.Sm' 24px, 'AvatarBaseSize.Md' 32px, 'AvatarBaseSize.Lg' 40px, 'AvatarBaseSize.Xl' 48px
+   * Defaults to AvatarBaseSize.Md
+   */
+  size?: AvatarBaseSize;
```

Comment:
Same question as AvatarNetwork is there a way we could allow for AvatarNetworkSize enum? Not sure if we want to mix enum names and components? Do you have any thoughts on this @brad-decker? If we update the engineer will have to use something like

```
<AvatarToken size={AvatarBaseSize.Lg} />
```

Which I'm not a huge fan of...